From f6ffeddfc78199b5da2dc5d51a47cbc4426595f4 Mon Sep 17 00:00:00 2001
From: Nick Kralevich <nnk@google.com>
Date: Sun, 17 Apr 2016 20:02:04 -0700
Subject: [PATCH 2/8] Maybe fix mac build

Mac builds don't have posix_fallocate, so only define this for
Linux builds.

Fixes the following build error:

  external/sqlite/dist/sqlite3.c:27063:42: error: use of undeclared
  identifier 'posix_fallocate'
    { "fallocate",    (sqlite3_syscall_ptr)posix_fallocate,  0 },

Change-Id: Ie530861069d2efbc55bbe1e4e622094798cbe869
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 dist/Android.mk | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/dist/Android.mk b/dist/Android.mk
index 7cf3659..aacdb4e 100644
--- a/dist/Android.mk
+++ b/dist/Android.mk
@@ -12,7 +12,6 @@ LOCAL_PATH:= $(call my-dir)
 #   SQLITE_DEFAULT_AUTOVACUUM=1  causes the databases to be subject to auto-vacuum
 minimal_sqlite_flags := \
 	-DNDEBUG=1 \
-	-DHAVE_POSIX_FALLOCATE=1 \
 	-DHAVE_USLEEP=1 \
 	-DSQLITE_HAVE_ISNAN \
 	-DSQLITE_DEFAULT_JOURNAL_SIZE_LIMIT=1048576 \
@@ -30,6 +29,9 @@ minimal_sqlite_flags := \
 	-DSQLITE_OMIT_LOAD_EXTENSION \
 	-DSQLITE_DEFAULT_FILE_PERMISSIONS=0600
 
+minimal_linux_flags := \
+    -DHAVE_POSIX_FALLOCATE=1 \
+
 device_sqlite_flags := $(minimal_sqlite_flags) \
     -DSQLITE_ENABLE_ICU \
     -DUSE_PREAD64 \
@@ -45,6 +47,7 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(common_src_files)
 
 LOCAL_CFLAGS += $(device_sqlite_flags)
+LOCAL_CFLAGS_linux += $(minimal_linux_flags)
 
 LOCAL_SHARED_LIBRARIES := libdl
 
@@ -68,6 +71,7 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(common_src_files)
 LOCAL_LDLIBS += -lpthread -ldl
 LOCAL_CFLAGS += $(minimal_sqlite_flags)
+LOCAL_CFLAGS_linux += $(minimal_linux_flags)
 LOCAL_MODULE:= libsqlite
 LOCAL_SHARED_LIBRARIES += libicuuc-host libicui18n-host
 LOCAL_STATIC_LIBRARIES := liblog libutils libcutils
@@ -100,6 +104,7 @@ LOCAL_SHARED_LIBRARIES := libsqlite \
 LOCAL_STATIC_LIBRARIES := libicuandroid_utils
 
 LOCAL_CFLAGS += $(device_sqlite_flags)
+LOCAL_CFLAGS_linux += $(minimal_linux_flags)
 
 LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)
 
@@ -123,6 +128,7 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(common_src_files) shell.c
 LOCAL_CFLAGS += $(minimal_sqlite_flags) \
     -DNO_ANDROID_FUNCS=1
+LOCAL_CFLAGS_linux += $(minimal_linux_flags)
 
 # sqlite3MemsysAlarm uses LOG()
 LOCAL_STATIC_LIBRARIES += liblog
@@ -142,6 +148,7 @@ include $(BUILD_HOST_EXECUTABLE)
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(common_src_files)
 LOCAL_CFLAGS += $(minimal_sqlite_flags)
+LOCAL_CFLAGS_linux += $(minimal_linux_flags)
 LOCAL_MODULE:= libsqlite_static_minimal
 LOCAL_SDK_VERSION := 23
 include $(BUILD_STATIC_LIBRARY)
@@ -150,5 +157,6 @@ include $(BUILD_STATIC_LIBRARY)
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(common_src_files)
 LOCAL_CFLAGS += $(minimal_sqlite_flags)
+LOCAL_CFLAGS_linux += $(minimal_linux_flags)
 LOCAL_MODULE:= libsqlite_static_minimal
 include $(BUILD_HOST_STATIC_LIBRARY)
-- 
2.10.0

